packages/simcore-sdk/tests/integration/test_node_ports_common_filemanager.py

```diff
@@ -63,13 +97,127 @@ async def test_valid_upload_download(
     assert filecmp.cmp(download_file_path, file_path)
 
 
+@pytest.fixture
+def mocked_upload_file_raising_exceptions(mocker: MockerFixture):
+    mocker.patch(
+        "simcore_sdk.node_ports_common.filemanager.sync_local_to_s3",
+        autospec=True,
+        side_effect=RCloneFailedError,
+    )
+    mocker.patch(
+        "simcore_sdk.node_ports_common.filemanager._upload_file_to_link",
+        autospec=True,
+        side_effect=exceptions.S3TransferError,
+    )
+
+
+@pytest.mark.parametrize(
+    "file_size",
+    [
+        _file_size("10Mib"),
+        _file_size("151Mib"),
+    ],
+)
+async def test_failed_upload_is_properly_removed_from_storage(
+    node_ports_config: None,
+    create_file_of_size: Callable[[ByteSize], Path],
+    create_valid_file_uuid: Callable[[Path], SimcoreS3FileID],
+    s3_simcore_location: LocationID,
+    optional_r_clone: Optional[RCloneSettings],
+    file_size: ByteSize,
+    user_id: UserID,
+    mocked_upload_file_raising_exceptions: None,
+):
+    file_path = create_file_of_size(file_size)
+    file_id = create_valid_file_uuid(file_path)
+    with pytest.raises(exceptions.S3TransferError):
+        await filemanager.upload_file(
+            user_id=user_id,
+            store_id=s3_simcore_location,
+            store_name=None,
+            s3_object=file_id,
+            local_file_path=file_path,
+            r_clone_settings=optional_r_clone,
+        )
+    with pytest.raises(exceptions.S3InvalidPathError):
+        await filemanager.get_file_metadata(
+            user_id=user_id, store_id=s3_simcore_location, s3_object=file_id
+        )

```

Comment:
What about the case when uploading an already existing file (overwriting it). What happens if the upload fails? Is it acceptable to "lose" the file? I mean from the standpoint of S3 you have a different version or you see that the file was deleted. But from a platform standpoint what do we desire?

Reply:
ok, so that is now the middle version. in the next PR it will properly revert to the last version. For now I unified how the mechanism works.
filemanager handles the file uploads. RClone and using presigned links are just 2 different tools for achieving it. there is no need that the RClone implementation does things more smartly than the other tool.
But you are right in this PR, it still will delete, which is the same as what is already happenning now. next PR will fix it.